diff --git a/CHANGES.rst b/CHANGES.rst
index 60985e1..17c1a19 100644
--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -4,6 +4,7 @@ Version 2.3.0
 Unreleased
 
 -   Drop support for Python 3.8 and 3.9.
+-   Remove previously deprecated code.
 
 
 Version 2.2.0
diff --git a/src/itsdangerous/__init__.py b/src/itsdangerous/__init__.py
index ea55256..2ce2ef8 100644
--- a/src/itsdangerous/__init__.py
+++ b/src/itsdangerous/__init__.py
@@ -1,7 +1,3 @@
-from __future__ import annotations
-
-import typing as t
-
 from .encoding import base64_decode as base64_decode
 from .encoding import base64_encode as base64_encode
 from .encoding import want_bytes as want_bytes
@@ -19,20 +15,3 @@
 from .timed import TimestampSigner as TimestampSigner
 from .url_safe import URLSafeSerializer as URLSafeSerializer
 from .url_safe import URLSafeTimedSerializer as URLSafeTimedSerializer
-
-
-def __getattr__(name: str) -> t.Any:
-    if name == "__version__":
-        import importlib.metadata
-        import warnings
-
-        warnings.warn(
-            "The '__version__' attribute is deprecated and will be removed in"
-            " ItsDangerous 2.3. Use feature detection or"
-            " 'importlib.metadata.version(\"itsdangerous\")' instead.",
-            DeprecationWarning,
-            stacklevel=2,
-        )
-        return importlib.metadata.version("itsdangerous")
-
-    raise AttributeError(name)
