diff --git a/CHANGELOG.rst b/CHANGELOG.rst
index ee12217..218e7ba 100644
--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -21,6 +21,11 @@ Bug fixes:
 - Fix typing of ``error_essages`` argument to `marshmallow.fields.Field` (:pr:`1636`).
   Thanks :user:`repole` for reporting and :user:`dhruvildarji` for the PR.
 
+Other changes:
+
+- Add ``ipaddress.*`` to `marshmallow.Schema.TYPE_MAPPING` (:issue:`1695`).
+  Thanks :user:`liberforce` for the suggestion and :user:`dhruvildarji` for the PR.
+
 4.2.2 (2026-02-04)
 ------------------
 
diff --git a/src/marshmallow/schema.py b/src/marshmallow/schema.py
index 4288f05..24efa90 100644
--- a/src/marshmallow/schema.py
+++ b/src/marshmallow/schema.py
@@ -8,6 +8,7 @@ import datetime as dt
 import decimal
 import functools
 import inspect
+import ipaddress
 import json
 import operator
 import typing
@@ -290,6 +291,10 @@ class Schema(metaclass=SchemaMeta):
         dt.date: ma_fields.Date,
         dt.timedelta: ma_fields.TimeDelta,
         decimal.Decimal: ma_fields.Decimal,
+        ipaddress.IPv4Address: ma_fields.IPv4,
+        ipaddress.IPv6Address: ma_fields.IPv6,
+        ipaddress.IPv4Interface: ma_fields.IPv4Interface,
+        ipaddress.IPv6Interface: ma_fields.IPv6Interface,
     }
     #: Overrides for default schema-level error messages
     error_messages: dict[str, str] = {}
