diff --git a/tests/tests_asyncio.py b/tests/tests_asyncio.py
index 250e658..877c6da 100644
--- a/tests/tests_asyncio.py
+++ b/tests/tests_asyncio.py
@@ -131,3 +131,20 @@ async def test_gather(capsys):
     _, err = capsys.readouterr()
     assert '30/30' in err
     assert res == list(range(0, 30 * 2, 2))
+
+
+async def raise_exc(i):
+    if i == 1:
+        raise ValueError("test")
+    return i * 2
+
+
+@mark.asyncio
+async def test_gather_exceptions(capsys):
+    """Test asyncio gather return_exceptions"""
+    res = await gather(*map(raise_exc, range(3)), return_exceptions=True)
+    _, err = capsys.readouterr()
+    assert '3/3' in err
+    assert isinstance(res[1], ValueError)
+    assert res[0] == 0
+    assert res[2] == 4
diff --git a/tqdm/asyncio.py b/tqdm/asyncio.py
index b32e767..72cef23 100644
--- a/tqdm/asyncio.py
+++ b/tqdm/asyncio.py
@@ -68,12 +68,18 @@ class tqdm_asyncio(std_tqdm):
                        total=total, **tqdm_kwargs)
 
     @classmethod
-    async def gather(cls, *fs, loop=None, timeout=None, total=None, **tqdm_kwargs):
+    async def gather(cls, *fs, loop=None, timeout=None, total=None, return_exceptions=False,
+                     **tqdm_kwargs):
         """
         Wrapper for `asyncio.gather`.
         """
         async def wrap_awaitable(i, f):
-            return i, await f
+            try:
+                return i, await f
+            except Exception as e:  # noqa: BLE001
+                if return_exceptions:
+                    return i, e
+                raise
 
         ifs = [wrap_awaitable(i, f) for i, f in enumerate(fs)]
         res = [await f for f in cls.as_completed(ifs, loop=loop, timeout=timeout,
